pkg/cloudprovider/aws/instancetypes.go

```diff
@@ -31,34 +31,41 @@ import (
 )
 
 const (
-	instanceTypesCacheKey         = "types"
-	instanceTypeZonesCacheKey     = "zones"
-	instanceTypesAndZonesCacheTTL = 5 * time.Minute
+	InstanceTypesCacheKey                         = "types"
+	InstanceTypeZonesCacheKey                     = "zones"
+	InstanceTypesAndZonesCacheTTL                 = 5 * time.Minute
+	InsufficientCapacityErrorCacheTTL             = 45 * time.Second
+	InsufficientCapacityErrorCacheCleanupInterval = 5 * time.Minute
 )
 
 type InstanceTypeProvider struct {
 	ec2api         ec2iface.EC2API
 	subnetProvider *SubnetProvider
-	cache          *cache.Cache
+	// Has two entries: one for all the instance types and one for all zones; values cached *before* considering insufficient capacity errors
+	// from the unavailableOfferings cache
+	cache *cache.Cache
+	// key: <capacityType>:<instanceType>:<zone>, value: struct{}{}
+	unavailableOfferings *cache.Cache
 }
 
 func NewInstanceTypeProvider(ec2api ec2iface.EC2API, subnetProvider *SubnetProvider) *InstanceTypeProvider {
 	return &InstanceTypeProvider{
-		ec2api:         ec2api,
-		subnetProvider: subnetProvider,
-		cache:          cache.New(instanceTypesAndZonesCacheTTL, CacheCleanupInterval),
+		ec2api:               ec2api,
+		subnetProvider:       subnetProvider,
+		cache:                cache.New(InstanceTypesAndZonesCacheTTL, CacheCleanupInterval),
+		unavailableOfferings: cache.New(InsufficientCapacityErrorCacheTTL, InsufficientCapacityErrorCacheCleanupInterval),
 	}
 }
 
-// Get instance type options given the constraints
+// Get all instance type options (the constraints are only used for tag filtering on subnets, not for Requirements filtering)
 func (p *InstanceTypeProvider) Get(ctx context.Context, constraints *v1alpha1.Constraints) ([]cloudprovider.InstanceType, error) {
```

Comment:
Is it possible to do
```suggestion
func (p *InstanceTypeProvider) Get(ctx context.Context, provider *v1alpha1.AWS) ([]cloudprovider.InstanceType, error) {
```